File: C/Algorithms/eratosthenes.c
#include <stdio.h>
#include <string.h>

int main(int argc, char *argv[]) {
	char arg1[6];
	if (argc > 1) strcpy(arg1, argv[1]);

	int end = 10000;
	char primes[end];
	for (int i = 0; i < end; i++)
		primes[end] = 1;

	primes[0] = 0;
	primes[1] = 0;

	for (int n = 2; n < end; n++) {
		for (int multiple = n << 1; multiple < end; multiple += n) {
			primes[multiple] = 0;
		}
	}

	// Print primes if "print" arg passed
	if (argc == 2 && !strcmp(arg1, "print")) {
		for (int i = 0; i < end; i++) {
			if (primes[i]) printf("%d\n", i);
		}
	}

	return 0;
}

File: Rust/algorithms/eratosthenes.rs
use std::env;
use std::time::SystemTime;

fn main() {
	let end = 10_000;
	let mut primes = vec![true; end];

	let start = SystemTime::now();

	primes[0] = false;
	primes[1] = false;

	for n in 2..end {
		for multiple in ((n << 1)..end).step_by(n) {
			primes[multiple] = false;
		}
	}

	let time = SystemTime::now()
		.duration_since(start)
		.unwrap();

	// Print primes if "print" arg passed
	let args: Vec<String> = env::args().collect();
	if args.len() > 1 && args[1] == "print" {
		primes
			.iter()
			.enumerate()
			.filter(|(_, b)| -> bool {**b})
			.for_each(|(i, _)| println!("{}", i));
	}

	println!("Time taken: {} seconds", time.as_secs_f64());
}